doc/building.rst

```diff
@@ -4,99 +4,188 @@
+   This workaround may lead to other shell warnings.
 
-    export PATH=/usr/local/Cellar/qt5/5.x.y/bin:$PATH
+.. _`a known issue`: https://stackoverflow.com/a/63241724
 
-   Where ``x.y`` is the current version of Qt 5 that brew has installed
-   on your machine.
-8. Install qtkeychain from here:  git clone https://github.com/frankosterfeld/qtkeychain.git
-   make sure you make the same install prefix as later while building the client e.g.  -
-   ``DCMAKE_INSTALL_PREFIX=/Path/to/client-install``
+4. Install Homebrew packages:
 
-9. For compilation of the client, follow the :ref:`generic-build-instructions`.
+   .. code-block:: bash
 
-10. Install the Packages_ package creation tool.
+      % brew install git qt qtkeychain cmake openssl pkgconfig glib doxygen cmocka
 
-11. Enable git submodules:
-```
-$ cd desktop
-$ git submodule init
-$ git submodule update
-```
+5. Certain Homebrew packages are not automatically linked in places where
+   the build scripts can find them, so create a shell-profile script that
+   can load the necessary environmental variables when you run a build:
 
-12. Generate the build files:
-```
-$ cd build
-$ cmake .. -DCMAKE_INSTALL_PREFIX=~/nextcloud-desktop-client -DCMAKE_BUILD_TYPE=Debug
-```
+   .. code-block:: bash
 
-13. Compile and install:
-```
-$ make install
-```
+      % echo 'export OPENSSL_ROOT_DIR=$(brew --prefix openssl)' >> ~/.nextcloud_build_variables
+      % echo 'export QT_PATH=$(brew --prefix qt5)/bin' >> ~/.nextcloud_build_variables
+      % echo 'export Qt5LinguistTools_DIR=$(brew --prefix qt5)/lib/cmake/Qt5LinguistTools/' >> ~/.nextcloud_build_variables
+   
+   .. note:: The file ``~/.nextcloud_build_variables`` is just a suggestion for
+      convenience. You can use a different file or create an entire shell
+      script, but this way of doing things is the simplest to explain.
 
-   .. note:: Contrary to earlier versions, Nextcloud 1.7 and later are packaged
-             as a ``pkg`` installer. Do not call "make package" at any time when
-             compiling for OS X, as this will build a disk image, and will not
-             work correctly.
+6. Clone the Nextcloud repository to a convenient location, such as ``~/Repositories``:
+
+   .. code-block:: bash
+
+      % cd ~/Repositories
+
+   .. note:: The cloned repository can go basically anywhere your user account
+      has write access, though it should not go in a directory synced with another
+      cloud service (especially not iCloud Drive). ``~/Repositories`` is recommended
+      for tidiness and consistency.
+
+   .. code-block:: bash
+
+      % git clone <repository_url>
+
+   (See the above section on using GitHub for an explanation of what URL to use.)
+
+7. Create build directory:
+
+   .. code-block:: bash
+
+      % cd ~/Repositories/desktop
+      % mkdir build
+
+8. Generate the build files:
+
+   .. code-block:: bash
+
+      % source ~/.nextcloud_build_variables
+      % cd ~/Repositories/desktop/build
```

Comment:
Not wrong but unnecessary if you have taken step `7` already. Could then be simplified to `cd build`.

Reply:
It is better to be detailed.

Reply:
I mentioned this in https://github.com/nextcloud/desktop/issues/2893, but I put the full path in the `cd` each time because I want it to be less of an issue if the user goes back and does the steps out of order.